frontend/src/components/ConstraintForm/ClassAccordionItem.vue

```diff
@@ -0,0 +1,55 @@
+<script>
+import { AccordionItem, AccordionList } from 'vue3-rich-accordion';
+import SubjectAccordionItem from './SubjectAccordionItem.vue';
+import { useFormStore } from '@/stores/form';
+
+export default {
+  props: {
+    index: { type: Number, default: -1 },
+  },
+  setup(props) {
+    const formStore = useFormStore();
+
+    const accordionId = `${props.index}`;
+
+    return { formStore, accordionId };
+  },
+  components: {
+    AccordionList,
+    AccordionItem,
+    SubjectAccordionItem,
+  },
+};
+</script>
+
+<template>
+  <accordion-item
+    :id="accordionId"
+    :default-opened="accordionId in formStore.accordionState ? formStore.accordionState[accordionId] : true"
+    class="class-accordion"
+  >
+    <template #summary>
+      <input
+        v-model="formStore.form.classes[index].name"
+        @click.stop
```

Comment:
Why is this event handler here?

Reply:
because click inside input should not trigger accordion expand/shrink

Reply:
Then good job on finding a workaround but it is weird to me the lib works like that. Some of these libs allow changing the close/open trigger to only fire when clicking the expand icon. You can look into that if you have the time.

Reply:
yes, that is because whole header part of accordion is clickable... and it is probably not designed to put input element in it...